Content.Client/HealthAnalyzer/UI/HealthAnalyzerWindow.xaml.cs

```diff
@@ -1,76 +1,203 @@
+using System.Linq;
 using System.Text;
 using Content.Shared.Damage;
 using Content.Shared.Damage.Prototypes;
 using Content.Shared.FixedPoint;
 using Content.Shared.IdentityManagement;
 using Content.Shared.MedicalScanner;
+using Content.Shared.Mobs.Components;
 using Robust.Client.AutoGenerated;
 using Robust.Client.UserInterface.CustomControls;
 using Robust.Client.UserInterface.XAML;
+using Robust.Client.GameObjects;
+using Robust.Client.Graphics;
+using Robust.Client.UserInterface.Controls;
 using Robust.Shared.Prototypes;
+using Robust.Shared.Utility;
 
 namespace Content.Client.HealthAnalyzer.UI
 {
     [GenerateTypedNameReferences]
     public sealed partial class HealthAnalyzerWindow : DefaultWindow
     {
+        private readonly IEntityManager _entityManager;
+        private readonly SpriteSystem _spriteSystem;
+        private readonly IPrototypeManager _prototypes;
+
+        private bool _isSettledWidth = false;
+
+        private const int AnalyzerHeight = 430;
+        private const int AnalyzerMaxWidth = 500;
+
         public HealthAnalyzerWindow()
         {
             RobustXamlLoader.Load(this);
+
+            _entityManager = IoCManager.Resolve<IEntityManager>();
+            _spriteSystem = _entityManager.System<SpriteSystem>();
+            _prototypes = IoCManager.Resolve<IPrototypeManager>();
+
+            RootContainer.OnResized += OnRootContainerResized;
```

Comment:
This looks like a hack around trying to get a specific UI layout instead of handling arrangement.